app/src/org/commcare/engine/resource/AndroidResourceUtils.kt

```diff
@@ -0,0 +1,44 @@
+package org.commcare.engine.resource
+
+import org.commcare.android.resource.installers.MediaFileAndroidInstaller
+import org.commcare.resources.model.MissingMediaException
+import org.commcare.resources.model.Resource
+import org.javarosa.core.reference.InvalidReferenceException
+import org.javarosa.core.reference.ReferenceManager
+import java.io.File
+import java.io.IOException
+import java.util.*
+
+object AndroidResourceUtils {
+
+    // loops over all lazy resources and checks if one of them has the same local path as {@param problem} URI
+    @JvmStatic
+    fun ifUriBelongsToALazyResource(problem: MissingMediaException, lazyResources: Vector<Resource>): Boolean {
+        for (lazyResource in lazyResources) {
```

Comment:
Not sure where this list is populated from, but heads up that iterating over resources is a really expensive operation on really large apps like CAS. This might need to get cached into a table which is indexed by the file URI